src/terminal/adapter/adaptDispatch.cpp

```diff
@@ -1318,7 +1323,7 @@ void AdaptDispatch::RequestChecksumRectangularArea(const VTInt id, const VTInt p
 {
     uint16_t checksum = 0;
     // If this feature is not enabled, we'll just report a zero checksum.
-    if constexpr (Feature_VtChecksumReport::IsEnabled())
+    if (Feature_VtChecksumReport::IsEnabled() && _vtChecksumReportEnabled)
```

Comment:
I thought the compiler would complain if you put a constexpr bool into the same if condition as a regular bool. Are the warnings broken for the adapter project? Hmm...

(It doesn't really matter though because `if constexpr` is mostly syntactic sugar. The compiler can also remove all other constant if conditions if optimizations are enabled.)

If the feature now has a bool member, do we still need the feature flag?

Reply:
Hmm... For WT, probably not. I'm guessing we'll still want it for `WindowsInbox` though? I'll update the feature flag to reflect that.

CC @DHowett

Reply:
Though I guess `_vtChecksumReportEnabled` is `false` by default so it'd be the same either way haha. Imma just remove the flag altogether 😅

Reply:
I think we will want to keep it turned off for WindowsInbox. Therefore: keep the `if constexpr` and wrap it around the `if (not constant expression)`, or flip the sense of the constexpr if to reduce nesting.

FWIW: I prefer that we keep all velocity-related `if constexpr` separate and on their own condition lines. It makes it easier to cleanup, and easier to diff when cleaned up.